src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -354,15 +359,48 @@ CreateDistributedTable(Oid relationId, Var *distributionColumn, char distributio
 	 * given relation is involved, then we undistribute the relation and finally
 	 * we re-create dropped foreign keys at the end of this function.
 	 */
-	List *fKeyCreationCommandsRelationInvolved = NIL;
+	List *originalForeignKeyRecreationCommands = NIL;
 	if (IsCitusTableType(relationId, CITUS_LOCAL_TABLE))
 	{
 		/* store foreign key creation commands that relation is involved */
-		fKeyCreationCommandsRelationInvolved =
-			GetFKeyCreationCommandsRelationInvolved(relationId);
+		originalForeignKeyRecreationCommands =
+			GetFKeyCreationCommandsRelationInvolvedWithTableType(relationId,
+																 INCLUDE_ALL_TABLE_TYPES);
 		relationId = DropFKeysAndUndistributeTable(relationId);
```

Comment:
question, should we switch to local execution here too (related #4489) ?

Reply:
I think yes,

```sql

create table ref2 (a int primary key);
create table local (X int references ref2(a), y int);
alter table local add constraint pk primary key (x);
create table local2 (x int references ref2(a), y int references local (x));
select create_reference_table('ref2');
ERROR:  cannot switch local execution status from local execution disabled to local execution enabled since it can cause visibility problems in the current transaction
DETAIL:  When adding a foreign key from a local table to a reference table, Citus applies a conversion to all the local tables in the foreign key graph
```

Reply:
I'm guessing that already happens when sending a drop constraint command to a Citus local table

Reply:
True.
When dropping citus local table, we will always use local execution unless local execution is disabled.
If local execution is disabled, then `CreateCitusLocalTable` would already error out.

But to be on the safe side, does it make sense to set local execution to required  ?

Reply:
It definitely can't hurt.